eval/eval.go

```diff
@@ -115,8 +115,13 @@ func RotateEnvironment(
 	providers ProviderLoader,
 	environments EnvironmentLoader,
 	execContext *esc.ExecContext,
+	paths []string,
```

Comment:
Might want this to be a list of parsed paths (e.g. `[]resource.PropertyPath` or something). At the very least we should parse these for validation and normalization prior to adding them to the et.